apps/notification/receivers/entry_comment.py

```diff
@@ -89,3 +95,49 @@ def create_entry_commit_text_notification(sender, instance, created, **kwargs):
         )
 
     transaction.on_commit(lambda: send_notifications_for_commit(comment.pk, meta))
+
+
+@receiver(m2m_changed, sender=Lead.assignee.through)
+def lead_assignment_signal(sender, instance, action, **kwargs):
+    pk_set = kwargs.get('pk_set', [])
+    if action == 'post_add' and pk_set:
+        for receiver in pk_set:
+            Assignment.objects.create(
+                content_object=instance,
+                created_for_id=receiver,
+                project=instance.project,
+                created_by=instance.created_by,
```

Comment:
We should use `request.user` instead of an instance.created_by. Take a condition where a user1 changes a lead create by user2.

For getting the request. You can either use a library module or a custom code.
- https://pypi.org/project/django-currentuser/
- https://github.com/IFRCGo/go-api/blob/develop/middlewares/middlewares.py#L12-L48